User_Story#1/SortingApp/ViewModel/CollectionSortingViewModel.cs

```diff
@@ -0,0 +1,271 @@
+using System.Runtime.CompilerServices;
+using System.Windows.Forms;
+using BusinessLayer;
+using BusinessLayer.DataParser;
+using BusinessLayer.Reader;
+using BusinessLayer.Validator;
+using BusinessLayer.Writer;
+using Helper;
+
+namespace ViewModel
+{
+    /// <summary>
+    /// Represents a ViewModel layer with all fields from View and logic how to handle requests from UI.
+    /// </summary>
+    /// <owner>Anton Petrenko</owner>
+    /// <typeparam name="T">Certain input type.</typeparam>
+    /// <seealso cref="System.ComponentModel.INotifyPropertyChanged" />
+    public class CollectionSortingViewModel<T> : INotifyPropertyChanged where T : IComparable
+    {
+        /// <summary>
+        /// The handler.
+        /// </summary>
+        /// <owner>Anton Petrenko</owner>
+        private readonly CollectionSortHandler<T> handler;
+
+        /// <summary>
+        /// Stores the error message.
+        /// </summary>
+        /// <owner>Anton Petrenko</owner>
+        private string message;
+
+        /// <summary>
+        /// The path for the source.
+        /// </summary>
+        /// <owner>Anton Petrenko</owner>
+        private string path;
+
+        /// <summary>
+        /// Stores the read command.
+        /// </summary>
+        /// <owner>Anton Petrenko</owner>
+        private RelayCommand read;
+
+        /// <summary>
+        /// Stores the save command.
+        /// </summary>
+        /// <owner>Anton Petrenko</owner>
+        private RelayCommand save;
+
+        /// <summary>
+        /// Stores the sort command.
+        /// </summary>
+        /// <owner>Anton Petrenko</owner>
+        private RelayCommand sort;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="CollectionSortingViewModel{T}"/> class.
+        /// </summary>
+        /// <owner>Anton Petrenko</owner>
+        public CollectionSortingViewModel()
+        {
+            this.handler = new CollectionSortHandler<T>
+                (new DataReader(new LocalFileValidator()), new DataWriter<T>(), new ArrayParser<T>());
+            this.UnSortedCollectionOfNumbers = new ObservableCollection<T>();
+            this.SortedCollectionOfNumbers = new ObservableCollection<T>();
+        }
+
+        /// <summary>
+        /// Gets the possibility of saving array.
+        /// </summary>
+        /// <param name="obj">The object.</param>
+        /// <returns></returns>
+        /// <owner>Anton Petrenko</owner>
+        private bool GetCanSaveArray(object obj)
+        {
+            return this.SortedCollectionOfNumbers.Any();
+        }
+
+        /// <summary>
+        /// Gets the can sort array.
+        /// </summary>
+        /// <param name="obj">The object.</param>
+        /// <returns></returns>
+        /// <owner>Anton Petrenko</owner>
+        private bool GetCanSortArray(object obj)
+        {
+            return (this.UnSortedCollectionOfNumbers.Any() && this.SortType is SortType valueOfSortType);
```

Comment:
Don't see a reason to store this.SortType in local variable valueOfSortType as it is not used further.
Also I don't see why did you add this casting if this property is already with that type....